src/app/modules/color-scheme/color-bar/color-bar.component.html

```diff
@@ -0,0 +1,5 @@
+<div *ngFor="let j of colorIndex" class="{{colorScheme.type}}" (click)="colorChanged([schemeIndex, j])" 
```

Comment:
Iterate over the scheme.colors instead. You can access the index with `*ngFor="...; index as i"`
You might also want to do ngIf based on discrete or gradient (ngIf supports else blocks)